Cmd line

jlchang opened the pull request:
Um, my first pull request - all feedback is appreciated!

one request, please take a look at the attribution to the Seurat tutorial that was written for the "process_data" function. Happy to make a more explicit/expansive acknowledgement or follow a style guide.

jlchang committed:
early draft, loads data, generates some initial plots

has lots of cruft from borrowed code to remind me of things to include later
still needs to incorporate more central data processing commands
to meet due date, will stop trying to generate all the QC graphs
and concentrate on central functionality first
then parameterizing the central functions
finally generating QC plots
(hope this attempt at multi-line commit messages works)

jlchang committed:
added rest of cell 1 as comments

jlchang committed:
comment out TSNEPlot command, trying cluster one marker discovery

jlchang committed:
added in remaining cells

jlchang committed:
first draft

jlchang committed:
move non-Seurat tutorial commands to separate function

jlchang committed:
added comments

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+# process_data function follows Seurat tutorial for processing PBMC data set made available by 10x Genomics
+# dataset https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz
+# Seurat tutorial https://satijalab.org/seurat/pbmc3k_tutorial.html
+process_data <- function(input.data){
```
This function can used more comments on what is happening internally.

jlchang replied:
description of implementation per Seurat tutorial site added as comment to describe the implementation

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+# Seurat tutorial https://satijalab.org/seurat/pbmc3k_tutorial.html
+process_data <- function(input.data){
+	dense.size <- object.size(x = as.matrix(x = input.data))
+	print(dense.size)
```
Dense size can be removed you are not using it in the rest of the code. Remove lines 16 and 17. This is a point in the tutorials about how different representations in memory of the data are more or less efficient.

jlchang replied:
removed

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	dense.size <- object.size(x = as.matrix(x = input.data))
+	print(dense.size)
+	seuratObj <- CreateSeuratObject(raw.data = input.data, min.cells = 3, min.genes = 200, project = "10X_data")
+	print(seuratObj)
```
min.cells and min.genes can be optional arguments.

TimothyTickle replied:
move print statements to logging using the logging class.

jlchang replied:
Comment added indicating TODO for parameterization
Comment added setting TODO for moving from print statements to logging

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	par(mfrow = c(1, 2))
+	GenePlot(object = seuratObj, gene1 = "nUMI", gene2 = "percent.mito")
+	GenePlot(object = seuratObj, gene1 = "nUMI", gene2 = "nGene")
+	seuratObj <- FilterCells(object = seuratObj, subset.names = c("nGene", "percent.mito"), low.thresholds = c(200, -Inf), high.thresholds = c(2500, 0.05))
```
lines 20 - 28 could be a separate block of code for filtering. Thinking how to reduce this large chunk of code.

TimothyTickle replied:
FilterCells can be parameterized as well. For now you can parameterize low.thresholds and high.thresholds into 4 arguments.

jlchang replied:
Comment added indicating TODO for parameterization
Does "separate block of code" mean a separate function or just visual separation (whitespace + comments)?

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	GenePlot(object = seuratObj, gene1 = "nUMI", gene2 = "nGene")
+	seuratObj <- FilterCells(object = seuratObj, subset.names = c("nGene", "percent.mito"), low.thresholds = c(200, -Inf), high.thresholds = c(2500, 0.05))
+	seuratObj <- NormalizeData(object = seuratObj, normalization.method = "LogNormalize", scale.factor = 10000)
+	seuratObj <- FindVariableGenes(object = seuratObj, mean.function = ExpMean, dispersion.function = LogVMR, x.low.cutoff = 0.0125, x.high.cutoff = 3, y.cutoff = 0.5)
```
x.low.cutoff, x.high.cuttoff, and y.cuttoff should be optional arguments

jlchang replied:
Comment added indicating TODO for parameterization

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	seuratObj <- FilterCells(object = seuratObj, subset.names = c("nGene", "percent.mito"), low.thresholds = c(200, -Inf), high.thresholds = c(2500, 0.05))
+	seuratObj <- NormalizeData(object = seuratObj, normalization.method = "LogNormalize", scale.factor = 10000)
+	seuratObj <- FindVariableGenes(object = seuratObj, mean.function = ExpMean, dispersion.function = LogVMR, x.low.cutoff = 0.0125, x.high.cutoff = 3, y.cutoff = 0.5)
+	length(x = seuratObj@var.genes)
```
Remove this line or print out the genes in logging. Length is not being used.

jlchang replied:
removed unneeded line

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	seuratObj <- FindVariableGenes(object = seuratObj, mean.function = ExpMean, dispersion.function = LogVMR, x.low.cutoff = 0.0125, x.high.cutoff = 3, y.cutoff = 0.5)
+	length(x = seuratObj@var.genes)
+	seuratObj <- ScaleData(object = seuratObj, vars.to.regress = c("nUMI", "percent.mito"))
+	seuratObj <- RunPCA(object = seuratObj, pc.genes = seuratObj@var.genes, do.print = TRUE, pcs.print = 1:5, genes.print = 5)
```
pcs.print and genes.print can be optional arguments

jlchang replied:
Comment added indicating TODO for parameterization

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	PrintPCA(object = seuratObj, pcs.print = 1:5, genes.print = 5, use.full = FALSE)
+	viz <- VizPCA(object = seuratObj, pcs.use = 1:2)
+	print(viz)
+	pca <- PCAPlot(object = seuratObj, dim.1 = 1, dim.2 = 2)
```
34 -37 are printing/plotting about principle components derived from PCA. Make sure pcs.print in the Run PCA and the above functions for PCs are coordinated. So pcs.print in PrintPCA should print the same PCs as Run. VIzPCA should also viz the same PCs and PCAPlot should print each of the PCs from pcs.print at least once as one of the dimensions in one of the plots. You do not have to do the full combinations of 1-5 choose 2 to plot in the X-Y of the plot but you can for instance plot 1v2, 2v3, 3v4, 4v5 and get mostly what people will want with the PCAPlot functionality.

TimothyTickle replied:
All the PCA functionality could also be a separate function

jlchang replied:
Indicated in comment to move PCA commands to new function

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	seuratObj <- ProjectPCA(object = seuratObj, do.print = FALSE)
+	seuratObj <- JackStraw(object = seuratObj, num.replicate = 100, display.progress = FALSE)
+	JackStrawPlot(object = seuratObj, PCs = 1:12)
+	jackstraw <- JackStrawPlot(object = seuratObj, PCs = 1:12)
```
will need to make the JackStraw PCs an optional argument

jlchang replied:
Comment added indicating TODO for parameterization
for JackStraw function num.replicate = 100, should num.replicate also be parameterized?

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	print(jackstraw)
+	elbow <- PCElbowPlot(object = seuratObj)
+	print(elbow)
+	seuratObj <- FindClusters(object = seuratObj, reduction.type = "pca", dims.use = 1:10, resolution = 0.6, print.output = 0, save.SNN = TRUE)
```
dims.use should be an optional argument.

jlchang replied:
Comment added indicating TODO for parameterization

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	print(elbow)
+	seuratObj <- FindClusters(object = seuratObj, reduction.type = "pca", dims.use = 1:10, resolution = 0.6, print.output = 0, save.SNN = TRUE)
+	PrintFindClustersParams(object = seuratObj)
+	seuratObj <- RunTSNE(object = seuratObj, dims.use = 1:10, do.fast = TRUE)
```
dims.use should be an argument and should be the same as FindClusters?

jlchang replied:
Comment indicating same value should be used for both functions

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	cluster1.markers <- FindMarkers(object = seuratObj, ident.1 = 1, min.pct = 0.25)
+	print(x = head(x = cluster1.markers, n = 5))
+	# find all markers distinguishing cluster 5 from clusters 0 and 3
+	cluster5.markers <- FindMarkers(object = seuratObj, ident.1 = 5, ident.2 = c(0, 3), min.pct = 0.25)
```
Why are these particular clusters selected in lines 50-53?

jlchang replied:
Will find out - added TODO to either comment on rationale and/or parameterize

jlchang replied:
demonstration code, removing to leave just FindAllMarkers

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	# only the positive ones
+	seuratObj.markers <- FindAllMarkers(object = seuratObj, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
+	seuratObj.markers %>% group_by(cluster) %>% top_n(2, avg_logFC)
+	cluster1.markers <- FindMarkers(object = seuratObj, ident.1 = 0, thresh.use = 0.25, test.use = "roc", only.pos = TRUE)
```
thresh.use should be an optional argument.

jlchang replied:
Comment added indicating TODO for parameterization

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+	print(trained_classifier) 
+	ranked_list = trained_classifier$importance[order(-trained_classifier$importance[,'MeanDecreaseAccuracy']),]
+	dput(rownames(ranked_list)[1:300]) 
+	FINAL_LIST = rownames(ranked_list)[1:100]
```
line 73 and 74 should use optional parameters for the subsetting ranges.

jlchang replied:
Comment added indicating TODO for parameterization

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+                              action="store",
+                              dest="input_dir",
+                              metavar="Input_Directory",
+                              help=paste("Input directory for analysis.",
```
this is a 10x output directory, include that information in the description.

jlchang replied:
Updated description of input to reflect expected format

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+if(!file.exists(args_parsed$input_dir[1])){
+    error_message <- paste0("Provided input directory, ",
+							args_parsed$input_dir[1], ",  does not exist.",
+    						"Please check your input and try again")
```
Make spacing consistent.in 103-105. If you are using tabs please do not use tabs in any code. Use other spacing.

jlchang replied:
will clean up tabs and use spaces instead

TimothyTickle commented on RF_PBMC.R:
```diff
@@ -0,0 +1,119 @@
+}
+
+
+# Manage inputs
```
please add more of a comment here.

jlchang replied:
comment updated to be more descriptive

TimothyTickle reviewed (changes requested):
Did a first pass on the review. Mostly cosmetic for ease of use and maintenance :-)

jlchang committed:
minor fixes and comments for next steps from PR

in case we need to release before implementing all the suggestions from the PR is completed

jlchang committed:
remove tabs, replace

jlchang committed:
freeze to test output before making structural changes

jlchang committed:
set seed (hard coded for now)to  test output reproducibility

jlchang committed:
cleaned up per lintr standards

jlchang committed:
clean up FindMarker code
